modules/api_reference/tests/src/FunctionalJavascript/OpenApiUploadTest.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+namespace Drupal\Tests\devportal_api_reference\FunctionalJavascript;
+
+use Drupal\FunctionalJavascriptTests\DrupalSelenium2Driver;
+use Drupal\FunctionalJavascriptTests\JavascriptTestBase;
+use Drupal\Tests\file\Functional\FileFieldCreationTrait;
+use Drupal\Tests\TestFileCreationTrait;
+
+/**
+ * @group devportal
+ * @group api_reference
+ */
+class OpenApiUploadTest extends JavascriptTestBase {
+
+  use FileFieldCreationTrait;
+  use TestFileCreationTrait;
+
+  protected const TITLE_NAME = 'title[0][value]';
+  protected const VERSION_NAME = 'field_version[0][value]';
+  protected const FILEFIELD_NAME = 'files[field_source_file_0]';
+
+  protected static $modules = [
+    'devportal_api_reference',
+    'block',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct($name = NULL, array $data = [], string $dataName = '') {
+    $this->minkDefaultDriverClass = DrupalSelenium2Driver::class;
+    parent::__construct($name, $data, $dataName);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->drupalPlaceBlock('page_title_block');
+    $this->drupalPlaceBlock('local_tasks_block');
+    $this->drupalPlaceBlock('local_actions_block');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function createScreenshot($filename_prefix = '', $set_background_color = TRUE) {
```

Comment:
Why do not you have this in a module specific JavascriptTestBase (abstract) parent class? I assume there will be more JS tests where you will use this.

Reply:
At the moment I have this one test, and I am not sure if a different one will be needed, since this module is quite feature complete at the moment. But when a new feature requests will emerge, I will refactor parts of this class into a base class or a trait, depending on the situation.